src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -314,16 +318,20 @@ private SynthesizedReadOnlyListTypeSymbol(SourceModuleSymbol containingModule, s
                     this,
                     ((MethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Collections_Generic_IEnumerable_T__GetEnumerator)!).AsMember(iEnumerableT),
                     generateGetEnumeratorT));
-            addProperty(membersBuilder,
-                new SynthesizedReadOnlyListProperty(
-                    this,
-                    ((PropertySymbol)compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IReadOnlyCollection_T__Count)!).AsMember(iReadOnlyCollectionT),
-                    generateCount));
-            addProperty(membersBuilder,
-                new SynthesizedReadOnlyListProperty(
-                    this,
-                    ((PropertySymbol)((MethodSymbol)compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IReadOnlyList_T__get_Item)!).AssociatedSymbol).AsMember(iReadOnlyListT),
-                    generateIndexer));
+            if ((PropertySymbol?)compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IReadOnlyCollection_T__Count) is { } wellKnownReadOnlyCollectionCount
+                && (MethodSymbol?)compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IReadOnlyList_T__get_Item) is { } wellKnownReadOnlyListIndexer)
```

Comment:
Perhaps `if (interfacesAreNotMissing)`.

Reply:
Do we need variables here, or can we simply add the members unless the containing interfaces are missing? For instance, if we capture a `bool hasReadOnlyInterfaces` early in the constructor, we could use that in several places, including here:
```csharp
if (hasReadOnlyInterfaces)
{
    addProperty(...);
    addProperty(...);
```